### PR TITLE
Add Yorkie Helm Charts

### PR DESCRIPTION
**Pull request by krapie**
**What this PR does / why we need it**:

Add Yorkie Helm Charts for Yorkie server cluster and monitoring/ArgoCD addons to support Yorkie K8s cluster.

Added helm charts are:
- `yorkie-cluster`: Helm chart for Yorkie server cluster
- `yorkie-monitoring`: Helm chart for Yorkie server cluster monitoring system with Prometheus and Grafana
- `yorkie-argocd`: Helm chart for Yorkie server cluster continuous delivery system with ArgoCD

Helm charts are located under: `/charts` directory with `README.md` for helm charts guide.

Also, helm charts repository will be self-hosted on GitHub Pages: https://yorkie-team.github.io/yorkie/helm-charts (or https://yorkie.dev/yorkie/helm-charts/) via `gh-pages` branch.

For this, [chart-release-action](https://github.com/helm/chart-releaser-action) GitHub action will be added. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: charts/yorkie-argocd/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+apiVersion: v2
+name: yorkie-argocd
+description: yorkie-argocd provides GitOps based continuous delivery for yorkie cluster
+type: application
+icon: https://raw.githubusercontent.com/yorkie-team/yorkie-team.github.io/main/public/favicon-512x512.png
+maintainers:
+  - name: hackerwins
+    email: youngteac.hong@navercorp.com
```
Comment (hackerwins) [inline charts/yorkie-argocd/Chart.yaml:8]:
```suggestion
    email: susukang98@gmail.com
```

File: charts/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Helm Charts
+
+This directory contains the Helm charts for Yorkie.
+
+## Charts
+
+- [yorkie-cluster](./yorkie-cluster): Helm chart for Yorkie server cluster
+- [yorkie-monitoring](./yorkie-monitoring): Helm chart for Yorkie server cluster monitoring system with Prometheus and Grafana
+- [yorkie-argocd](./yorkie-argocd): Helm chart for Yorkie server cluster continuous delivery system with ArgoCD
```
Comment (hackerwins) [inline charts/README.md:9]:
```suggestion
- [yorkie-cluster](./yorkie-cluster): Helm chart for Yorkie cluster
- [yorkie-monitoring](./yorkie-monitoring): Helm chart for Yorkie cluster monitoring system with Prometheus and Grafana
- [yorkie-argocd](./yorkie-argocd): Helm chart for Yorkie cluster continuous delivery system with ArgoCD
```

File: charts/yorkie-argocd/README.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+## Prerequisites
+
+- Kubernetes 1.24+
+- Helm 3+
+
+## Get Helm Repository Info
+
+```bash
+helm repo add yorkie https://yorkie-team.github.io/yorkie/helm-charts
+helm repo update
```
Comment (hackerwins) [inline charts/yorkie-argocd/README.md:14]:
```suggestion
$ helm repo add yorkie https://yorkie-team.github.io/yorkie/helm-charts
$ helm repo update
```

File: charts/yorkie-argocd/README.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+```
+
+_See [`helm repo`](https://helm.sh/docs/helm/helm_repo/) for command documentation._
+
+## Install Helm Chart
+
+```bash
+# Install yorkie argocd helm chart
+helm install [RELEASE_NAME] yorkie/yorkie-argocd -n argocd --create-namespace
```
Comment (hackerwins) [inline charts/yorkie-argocd/README.md:23]: What value should I put in RELEASE_NAME?

File: charts/yorkie-argocd/README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+# Update admin password
+## bcrypt(yorkie)=$2a$12$fJoZj9CnNyD5Yfi02nZh7.XcH4Ds9M.oftQOQDP5oytyra9cP6Dny
+kubectl -n argocd patch secret argocd-secret \
+  -p '{"stringData": {
+    "admin.password": "$2a$12$fJoZj9CnNyD5Yfi02nZh7.XcH4Ds9M.oftQOQDP5oytyra9cP6Dny",
+    "admin.passwordMtime": "'$(date +%FT%T%Z)'"
```
Comment (hackerwins) [inline charts/yorkie-argocd/README.md:30]: What is the actual value of `admin.password`?

File: charts/yorkie-cluster/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+          - containerPort: {{ .Values.yorkie.ports.sdkPort }}
+          - containerPort: {{ .Values.yorkie.ports.profilingPort }}
+          - containerPort: {{ .Values.yorkie.ports.adminPort }}
+        livenessProbe:
+          grpc:
+            port: {{ .Values.yorkie.ports.sdkPort }}
+          initialDelaySeconds: 10
+        resources:
+          {{ toYaml .Values.yorkie.resources | nindent 12 }}
+
```
Comment (hackerwins) [inline charts/yorkie-cluster/templates/deployment.yaml:55]:
```suggestion

```

**Review by hackerwins — CHANGES_REQUESTED** [review]
Thanks for your contribution.

Some files seem to be dependent on AWS. Can we install Yorkie Cluster on another K8S?

File: charts/yorkie-argocd/README.md (hunk 3 above)
Comment (krapie) [reply charts/yorkie-argocd/README.md:14]: `$` can improve readability, but it's a bit cumbersome to copy/paste commands.

For example, if I want to copy/paste CRD deletion commands, I need to:
```bash
kubectl delete crd authorizationpolicies.security.istio.io
kubectl delete crd destinationrules.networking.istio.io
kubectl delete crd envoyfilters.networking.istio.io
kubectl delete crd gateways.networking.istio.io
kubectl delete crd istiooperators.install.istio.io
kubectl delete crd peerauthentications.security.istio.io
kubectl delete crd proxyconfigs.networking.istio.io
kubectl delete crd requestauthentications.security.istio.io
kubectl delete crd serviceentries.networking.istio.io
kubectl delete crd sidecars.networking.istio.io
kubectl delete crd telemetries.telemetry.istio.io
kubectl delete crd virtualservices.networking.istio.io
kubectl delete crd wasmplugins.extensions.istio.io
kubectl delete crd workloadentries.networking.istio.io
kubectl delete crd workloadgroups.networking.istio.io
```

but putting `$` in front of all these commands will make it hard to execute commands.

File: charts/yorkie-monitoring/values.yaml [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+# Configuration for external gateway (eg: AWS ALB)
+externalGateway:
+  alb:
+    enabled: false
+
+    certArn: arn:aws:acm:ap-northeast-2:031403140314:certificate/4ae32f71-39df-43b6-bf19-c0428827920a
+    # Use same host for yorkie cluster
+    apiHost: api.yorkie.dev
+  
+    grafanaPath: /grafana
```
[the remark this answers is not in the document]
Comment (krapie) [reply charts/yorkie-monitoring/values.yaml:123]: This `certArn` value is for the example, I modified some characters in existing `certArn`.

But, I will modify it more and it make it look more like sample `certArn`: `arn:aws:acm:ap-northeast-2:123412341234:certificate/1234-1234-1234-1234-1234`

File: charts/yorkie-argocd/README.md (hunk 4 above)
Comment (krapie) [reply charts/yorkie-argocd/README.md]: Anything you want to name it as alias of chart. (eg: yorkie-cluster-minikube, yorkie-cluster-aws)
For more information, follow: https://helm.sh/ko/docs/intro/using_helm/

File: charts/yorkie-argocd/README.md (hunk 5 above)
Comment (krapie) [reply charts/yorkie-argocd/README.md]: it's on the commend above: `bcrypt(yorkie)=$2a$12$fJoZj9CnNyD5Yfi02nZh7.XcH4Ds9M.oftQOQDP5oytyra9cP6Dny`

File: charts/yorkie-argocd/values.yaml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  destination:
+    server: https://kubernetes.default.svc
+    namespace: default
+
+# Configuration for external gateway (eg: AWS ALB)
+externalGateway:
+  alb:
+    enabled: false
+
+    certArn: arn:aws:acm:ap-northeast-2:031403140314:certificate/4ae32f71-39df-43b6-bf19-c0428827920a
```
[the remark this answers is not in the document]
Comment (krapie) [reply charts/yorkie-argocd/values.yaml]: This certArn value is for the example, I modified some characters in existing certArn.

But, I will modify it more and it make it look more like sample. (eg: arn:aws:acm:ap-northeast-2:123412341234:certificate/1234-1234-1234-1234-1234)

**Comment by krapie** [comment]
> Some files seem to be dependent on AWS. Can we install Yorkie Cluster on another K8S?

I have integrated ingress to allow users to select different types of ingress on various platform(eg: AWS, GKE, minikube).
By default, ingress uses nginx ingress controller, but if user enables alb option, ingress will use alb ingress controller with proper annotations.

**Review by hackerwins — APPROVED** [review]
LGTM. 👍

Let's improve this while deploying using helm charts.

**Pull request merged**